### PR TITLE
Higher recovery timeout in PersistentActorRecoveryTimeoutSpec

### LINKED ISSUE #20728: FAILED: PersistentActorRecoveryTimeoutSpec
https://jenkins.akka.io:8498/job/akka-multi-node-nightly2/373/consoleFull

```
[info] - should fail recovery if timeout is not met when recovering *** FAILED *** (3 seconds, 144 milliseconds)
[info]   java.util.concurrent.TimeoutException: Futures timed out after [3 seconds]
[info]   at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:219)
[info]   at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:223)
[info]   at scala.concurrent.Await$$anonfun$result$1.apply(package.scala:190)
[info]   at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:53)
[info]   at scala.concurrent.Await$.result(package.scala:190)
[info]   at akka.persistence.journal.SteppingInmemJournal$.step(SteppingInmemJournal.scala:30)
[info]   at akka.persistence.PersistentActorRecoveryTimeoutSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply$mcV$sp(PersistentActorRecoveryTimeoutSpec.scala:51)
[info]   at akka.persistence.PersistentActorRecoveryTimeoutSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(PersistentActorRecoveryTimeoutSpec.scala:43)
[info]   at akka.persistence.PersistentActorRecoveryTimeoutSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(PersistentActorRecoveryTimeoutSpec.scala:43)
[info]   at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:953)
[info]   at org.scalatest.Suite$class.withFixture(Suite.scala:1122)
[info]   at akka.testkit.AkkaSpec.withFixture(AkkaSpec.scala:57)
```

### PR DESCRIPTION
**Pull request by johanandren**
To not fail on slow/gc

Fixes #20728

### CODE REVIEW

File: akka-persistence/src/test/scala/akka/persistence/PersistentActorRecoveryTimeoutSpec.scala [hunk 1]
```diff
@@ -15,7 +15,7 @@ object PersistentActorRecoveryTimeoutSpec {
     SteppingInmemJournal.config(PersistentActorRecoveryTimeoutSpec.journalId).withFallback(
       ConfigFactory.parseString(
         """
-          |akka.persistence.journal.stepping-inmem.recovery-event-timeout=100ms
+          |akka.persistence.journal.stepping-inmem.recovery-event-timeout=500ms
```
Comment (patriknw) [inline akka-persistence/src/test/scala/akka/persistence/PersistentActorRecoveryTimeoutSpec.scala]: what happens if you make it even longer? will that slow down the test too much?

Comment (johanandren) [inline akka-persistence/src/test/scala/akka/persistence/PersistentActorRecoveryTimeoutSpec.scala]: That's what I was thinking, but it's just a single negative test case so maybe a second wouldn't hurt too much.

Comment (patriknw) [inline akka-persistence/src/test/scala/akka/persistence/PersistentActorRecoveryTimeoutSpec.scala]: +1

**Comment by patriknw** [comment]
LGTM

**Comment by drewhk** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM

**Pull request merged**